### PR TITLE
Jetpack Overlay - Refactor landscape layout and add RTL animation

### LINKED ISSUE #19172: [iOS] Update landscape layout
The iPhone landscape layout of the Jetpack overlay needs to be updated. Here are two possible proposals

<p align=center>
Screen Shot 2022-08-10 at 4 33 35 PM

Screen Shot 2022-08-10 at 4 38 14 PM
</p>

### PR DESCRIPTION
**Pull request by Gio2018**
Fixes #19172

LTR | RTL
-- | --
 | 

This PR tweaks the landscape layout of the Jetpack overlay to better fit the content in it. 
Additionally: 
- adds a RTL animation, so that the WP logo falls on the leading edge in both orientations.
- Fixes banners and badges font, changing it from `body` to `callout`
- Tweaks the top constraint of the dismiss button to account for the gripper view

To test:
Set a LTR language
- Checkout this branch and enable the `jetpackPoweredBottomSheet` feature flag
- Tap on any "Jetpack powered" banner or badge to show the overlay
- Rotate to landscape and make sure the content is properly shown
- Make sure the animation has the wp logo at the leading edge when it finishes

Set a RTL language
- Repeat the steps
Additionally, test dark/light mode and accessibility fonts

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by twstokes — COMMENTED** [review]
This looks great to me @Gio2018. The only thing that I noticed is that when comparing to the mocks, I think we need to come down on the banner height some.

| Mock | Current |
| - | - |
|  |  |

**Comment by Gio2018** [comment]
> This looks great to me @Gio2018. The only thing that I noticed is that when comparing to the mocks, I think we need to come down on the banner height some.
> 

Thanks @twstokes ! Done!

File: WordPress/Classes/ViewRelated/Jetpack/Branding/Banner/JetpackBannerView.swift [hunk 1]
```diff
@@ -38,7 +38,7 @@ class JetpackBannerView: UIView {
     }
 
     /// Preferred minimum height to be used for constraints
-    static let minimumHeight: CGFloat = 50
+    static let minimumHeight: CGFloat = 44
```
Comment (twstokes) [inline WordPress/Classes/ViewRelated/Jetpack/Branding/Banner/JetpackBannerView.swift:41]: Awesome! Looks like we've done a good job making this the source of truth. I tested all views I'm aware of with a banner and they appear to correctly reflect the change.

**Review by twstokes — APPROVED** [review]
LGTM @Gio2018! 🚀

**Pull request merged**